packages/netlify-cms-backend-git-gateway/src/implementation.ts

```diff
@@ -125,18 +152,26 @@ export default class GitGateway implements Implementation {
       this.backendType = null;
     }
 
-    this.authClient = window.netlifyIdentity
-      ? window.netlifyIdentity.gotrue
-      : new GoTrue({ APIUrl });
-    AuthenticationPage.authClient = this.authClient;
-
     this.backend = null;
+    AuthenticationPage.authClient = () => this.getAuthClient();
   }
 
   isGitBackend() {
     return true;
   }
 
+  async getAuthClient() {
+    if (this.authClient) {
+      return this.authClient;
+    }
+    await initPromise;
+    const authClient = window.netlifyIdentity
```

Comment:
Can you check if something like so would work here
```js
const authClient = await new Promise((resolve) => {
  if (window.netlifyIdentity) {
    window.netlifyIdentity.store.gotrue &&
      resolve(window.netlifyIdentity.store.gotrue);
    window.netlifyIdentity.on("init", () => {
      resolve(window.netlifyIdentity.gotrue);
    });
  } else {
    resolve(new GoTrue({ APIUrl: this.APIUrl }));
  }
});
```

Reply:
`window.netlifyIdentity.store.gotrue` is implemented as a getter (https://github.com/netlify/netlify-identity-widget/blob/05152b12587e5093940c004f647d59799daaa21f/src/netlify-identity.js#L45) so we would like to avoid accessing it before the widget was initialized.
Also I thing it is best to register to the `init` event as soon as possible